test/slicing/test_slice_combiner.py

```diff
@@ -11,23 +12,191 @@ class SliceCombinerTest(unittest.TestCase):
+        }
+        combiner_module = SliceCombinerModule()
+        combined_rep = combiner_module(outputs)
+        self.assertTrue(torch.allclose(combined_rep, torch.ones(batch_size, h_dim) * 3))
+
+    def test_average_reweighting_by_ind(self):
+        """Test average reweighting by ind (zeros in pred head)."""
+        batch_size = 4
+        h_dim = 20
+        num_classes = 2
+
+        outputs = {
+            "task_slice:a_ind_head": [torch.ones(batch_size, 2) * 10.0],
+            "task_slice:a_pred_transform": [torch.ones(batch_size, h_dim) * 4],
+            "task_slice:a_pred_head": [torch.zeros(batch_size, num_classes)],
+            "task_slice:b_ind_head": [torch.ones(batch_size, 2) * -10.0],
+            "task_slice:b_pred_transform": [torch.ones(batch_size, h_dim) * 2],
+            "task_slice:b_pred_head": [torch.zeros(batch_size, num_classes)],
+        }
+        combiner_module = SliceCombinerModule()
+        combined_rep = combiner_module(outputs)
+        self.assertTrue(torch.allclose(combined_rep, torch.ones(batch_size, h_dim) * 3))
+
+    def test_average_reweighting_by_pred_confidence(self):
+        """Test average reweighting by pred confidence (zeros in ind head)."""
+        batch_size = 4
+        h_dim = 20
+        num_classes = 2
+
+        outputs = {
+            "task_slice:a_ind_head": [torch.zeros(batch_size, 2)],
+            "task_slice:a_pred_transform": [torch.ones(batch_size, h_dim) * 4],
+            "task_slice:a_pred_head": [torch.ones(batch_size, num_classes) * 5],
+            "task_slice:b_ind_head": [torch.zeros(batch_size, 2)],
+            "task_slice:b_pred_transform": [torch.ones(batch_size, h_dim) * 2],
+            "task_slice:b_pred_head": [torch.ones(batch_size, num_classes) * 5],
+        }
+        combiner_module = SliceCombinerModule()
+        combined_rep = combiner_module(outputs)
+        self.assertTrue(torch.all(combined_rep == torch.ones(batch_size, h_dim) * 3))
+
+        # Changing sign of prediction shouldn't change confidence
+        outputs = {
+            "task_slice:a_ind_head": [torch.zeros(batch_size, 2)],
+            "task_slice:a_pred_transform": [torch.ones(batch_size, h_dim) * 4],
+            "task_slice:a_pred_head": [torch.ones(batch_size, num_classes) * -5],
+            "task_slice:b_ind_head": [torch.zeros(batch_size, 2)],
+            "task_slice:b_pred_transform": [torch.ones(batch_size, h_dim) * 2],
+            "task_slice:b_pred_head": [torch.ones(batch_size, num_classes) * 5],
+        }
+        combiner_module = SliceCombinerModule()
+        combined_rep = combiner_module(outputs)
+        self.assertTrue(torch.allclose(combined_rep, torch.ones(batch_size, h_dim) * 3))
+
+    def test_many_slices(self):
+        """Test combiner on 1000 synthetic generated slices."""
+
+        batch_size = 4
+        h_dim = 20
+        num_classes = 2
+
+        outputs = {}
+        for i in range(1000):
+            if i % 2 == 0:
```

Comment:
sooooo what's going on here? half super confident 4 predictions, half super confident 2 predictions? maybe add a note. and can we get away with 100?

Reply:
yes half confident in each way; more to test whether the model can handle the scale

Reply:
100 slices will work; it isn't much more costly to initialize more